kafka receiver panic: close of closed channel

**Describe the bug**
I use kafkareceiver to receive data. When running for a period of time (about 10s), the program will panic.

```
panic: close of closed channel

goroutine 201 [running]:
go.opentelemetry.io/collector/receiver/kafkareceiver.(*consumerGroupHandler).Setup(0xc0000ae880, 0x1b8eea0, 0xc000798200, 0xc001d50660, 0xc000798200)
        /Users/niaoge/go/pkg/mod/go.opentelemetry.io/collector@v0.9.1-0.20200831045722-c3f7dd5e40a3/receiver/kafkareceiver/kafka_receiver.go:131 +0x36
github.com/Shopify/sarama.newConsumerGroupSession(0x1b79980, 0xc001544280, 0xc00002a3f0, 0xc001684150, 0xc0000100c0, 0x3b, 0x1c4f, 0x1b74dc0, 0xc0000ae880, 0x0, ...)
        /Users/niaoge/go/pkg/mod/github.com/!shopify/sarama@v1.27.0/consumer_group.go:597 +0x464
github.com/Shopify/sarama.(*consumerGroup).newSession(0xc00002a3f0, 0x1b79980, 0xc0000ae840, 0xc0001d2610, 0x1, 0x1, 0x1b74dc0, 0xc0000ae880, 0x4, 0xc00004a280, ...)
        /Users/niaoge/go/pkg/mod/github.com/!shopify/sarama@v1.27.0/consumer_group.go:299 +0x2f7
github.com/Shopify/sarama.(*consumerGroup).Consume(0xc00002a3f0, 0x1b79980, 0xc0000ae840, 0xc0001d2610, 0x1, 0x1, 0x1b74dc0, 0xc0000ae880, 0x0, 0x0)
        /Users/niaoge/go/pkg/mod/github.com/!shopify/sarama@v1.27.0/consumer_group.go:169 +0x18b
go.opentelemetry.io/collector/receiver/kafkareceiver.(*kafkaConsumer).consumeLoop(0xc0001e0690, 0x1b79980, 0xc0000ae840, 0x1b74dc0, 0xc0000ae880, 0x8c2c25, 0xc000022630)
        /Users/niaoge/go/pkg/mod/go.opentelemetry.io/collector@v0.9.1-0.20200831045722-c3f7dd5e40a3/receiver/kafkareceiver/kafka_receiver.go:103 +0x9b
created by go.opentelemetry.io/collector/receiver/kafkareceiver.(*kafkaConsumer).Start
        /Users/niaoge/go/pkg/mod/go.opentelemetry.io/collector@v0.9.1-0.20200831045722-c3f7dd5e40a3/receiver/kafkareceiver/kafka_receiver.go:93 +0x169
```

--- a/receiver/kafkareceiver/kafka_receiver.go
+++ b/receiver/kafkareceiver/kafka_receiver.go
@@ -17,6 +17,7 @@ package kafkareceiver
 import (
 	"context"
 	"fmt"
+	"sync"
 
 	"github.com/Shopify/sarama"
 	"go.opencensus.io/stats"
@@ -108,7 +109,6 @@ func (c *kafkaConsumer) consumeLoop(ctx context.Context, handler *consumerGroupH
 			c.logger.Info("Consumer stopped", zap.Error(ctx.Err()))
 			return ctx.Err()
 		}
-		handler.ready = make(chan bool)
 	}
 }
 
@@ -122,6 +122,7 @@ type consumerGroupHandler struct {
 	unmarshaller Unmarshaller
 	nextConsumer consumer.TraceConsumer
 	ready        chan bool
+	readyCloser sync.Once
 
 	logger *zap.Logger
 }
@@ -129,8 +130,9 @@ type consumerGroupHandler struct {
 var _ sarama.ConsumerGroupHandler = (*consumerGroupHandler)(nil)
 
 func (c *consumerGroupHandler) Setup(session sarama.ConsumerGroupSession) error {
-	close(c.ready)
-
+	c.readyCloser.Do(func() {
+		close(c.ready)
+	})
 	statsTags := []tag.Mutator{tag.Insert(tagInstanceName, c.name)}
 	_ = stats.RecordWithTags(session.Context(), statsTags, statPartitionStart.M(1))
 	return nil

--- a/receiver/kafkareceiver/kafka_receiver.go
+++ b/receiver/kafkareceiver/kafka_receiver.go
@@ -122,7 +122,7 @@ type consumerGroupHandler struct {
 	unmarshaller Unmarshaller
 	nextConsumer consumer.TraceConsumer
 	ready        chan bool
-	readyCloser sync.Once
+	readyCloser  sync.Once
 
 	logger *zap.Logger
 }

--- a/receiver/kafkareceiver/kafka_receiver.go
+++ b/receiver/kafkareceiver/kafka_receiver.go
@@ -95,7 +95,7 @@ func (c *kafkaConsumer) Start(context.Context, component.Host) error {
 	return nil
 }
 
-func (c *kafkaConsumer) consumeLoop(ctx context.Context, handler sarama.ConsumerGroupHandler) error {
+func (c *kafkaConsumer) consumeLoop(ctx context.Context, handler *consumerGroupHandler) error {
 	for {
 		// `Consume` should be called inside an infinite loop, when a
 		// server-side rebalance happens, the consumer session will need to be
@@ -108,6 +108,7 @@ func (c *kafkaConsumer) consumeLoop(ctx context.Context, handler sarama.Consumer
 			c.logger.Info("Consumer stopped", zap.Error(ctx.Err()))
 			return ctx.Err()
 		}
+		handler.ready = make(chan bool)
 	}
 }
 
@@ -129,6 +130,7 @@ var _ sarama.ConsumerGroupHandler = (*consumerGroupHandler)(nil)
 
 func (c *consumerGroupHandler) Setup(session sarama.ConsumerGroupSession) error {
 	close(c.ready)
+
 	statsTags := []tag.Mutator{tag.Insert(tagInstanceName, c.name)}
 	_ = stats.RecordWithTags(session.Context(), statsTags, statPartitionStart.M(1))
 	return nil

--- a/receiver/kafkareceiver/kafka_receiver.go
+++ b/receiver/kafkareceiver/kafka_receiver.go
@@ -96,7 +96,7 @@ func (c *kafkaConsumer) Start(context.Context, component.Host) error {
 	return nil
 }
 
-func (c *kafkaConsumer) consumeLoop(ctx context.Context, handler *consumerGroupHandler) error {
+func (c *kafkaConsumer) consumeLoop(ctx context.Context, handler sarama.ConsumerGroupHandler) error {
 	for {
 		// `Consume` should be called inside an infinite loop, when a
 		// server-side rebalance happens, the consumer session will need to be

